crates/biome_cli/src/execute/traverse.rs

```diff
@@ -483,11 +505,16 @@ pub(crate) struct TraversalOptions<'ctx, 'app> {
     /// The approximate number of diagnostics the console will print before
     /// folding the rest into the "skipped diagnostics" counter
     pub(crate) remaining_diagnostics: &'ctx AtomicU32,
+
+    /// List of paths that should be processed
+    pub(crate) evaluated_paths: RwLock<FxHashSet<EvaluatedPath>>,
```

Comment:
Just to be sure: why do we need a `RWLock` here? Have we several processes that “evaluate” files? Or it is just for toggling if a file was fixed? In this last case, we could use an atomic bool?

Reply:
`TraversalOptions` is sent as a **reference** across multiple threads. So we need `RwLock` for two reasons:
- enable interior mutability when having a `&self`
- lock the resource when multiple threads want to update the list